pkg/steps/template.go

```diff
@@ -591,6 +598,38 @@ func waitForPodCompletionOrTimeout(ctx context.Context, podClient coreclientset.
 	}
 }
 
+func removeFinalizer(ctx context.Context, podClient coreclientset.PodInterface, pod *coreapi.Pod, finalizer string) {
+	if ctx.Err() != nil {
+		// give ourselves a bit of a grace period to remove the finalizer
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(context.Background(), 5*time.Minute)
+		defer cancel()
+	}
+
+	nextFinalizers := make([]string, 0, len(pod.ObjectMeta.Finalizers))
+	for _, entry := range pod.ObjectMeta.Finalizers {
+		if entry != finalizer {
+			nextFinalizers = append(nextFinalizers, entry)
+		}
+	}
+	if len(nextFinalizers) != len(pod.ObjectMeta.Finalizers) {
+		data, err := json.Marshal(map[string]interface{}{
+			"op":    "replace",
+			"path":  "/metadata/finalizers",
+			"value": nextFinalizers,
+		})
+		if err != nil {
+			log.Printf("Failed to marshal patch for removing finalizer %s from pod %s: %v", finalizer, pod.Name, err)
+			return
+		}
+		if _, err = podClient.Patch(ctx, pod.Name, types.JSONPatchType, data, meta.PatchOptions{}); err == nil {
+			log.Printf("Finished watching pod %s; removed finalizer %s", pod.Name, finalizer)
+		} else {
+			log.Printf("Finished watching pod %s; failed to remove finalizer %s: %v", pod.Name, finalizer, err)
```

Comment:
We need something else that removes this finalizers if ci-operator fails to do so for whatever reasons, otherwise we will have stuck nodes

Reply:
What something else would that be?

And narrowly for #1257, I'm comfortable with bf10feb3420588b5345cbc6f75986ab5e29a562a, instead of the parallel `Delete` workaround, so feel free to reject this PR if you don't want the guard for the existing, parallel `Delete`.

Reply:
I don't know, but until we have something, we can not merge this change. Maybe we could add it to the namespace reaper controller?

Reply:
Can you link me to that controller?

Reply:
https://github.com/openshift/ci-ns-ttl-controller

Its currently based on the assumption that the existence of pods indicates that a namespace is in use, so that would need to be changed so that deletionTImeStamp != nil is considered as non-existent pod as well

Reply:
Makes sense, even if we don't land this finalizer PR: openshift/ci-ns-ttl-controller#7